agent/fw_drupal8.c

```diff
@@ -632,20 +748,26 @@ NR_PHP_WRAPPER(nr_drupal8_module_handler) {
 
   ce = Z_OBJCE_P(*retval_ptr);
 
-  nr_drupal8_add_method_callback(ce, NR_PSTR("getimplementations"),
-                                 nr_drupal8_post_get_implementations TSRMLS_CC);
-  nr_drupal8_add_method_callback(ce, NR_PSTR("implementshook"),
-                                 nr_drupal8_post_implements_hook TSRMLS_CC);
-  /* Drupal 9.4 introduced a replacement method for getImplentations */
+  hook_attribute_instrumentation
+      = nr_drupal_hook_attribute_instrument(*retval_ptr);
+
+  if (!hook_attribute_instrumentation) {
+    nr_drupal8_add_method_callback(
+        ce, NR_PSTR("getimplementations"),
+        nr_drupal8_post_get_implementations TSRMLS_CC);
+    nr_drupal8_add_method_callback(ce, NR_PSTR("implementshook"),
+                                   nr_drupal8_post_implements_hook TSRMLS_CC);
+    /* Drupal 9.4 introduced a replacement method for getImplentations */
 #if ZEND_MODULE_API_NO >= ZEND_8_0_X_API_NO \
     && !defined OVERWRITE_ZEND_EXECUTE_DATA
-  nr_drupal8_add_method_callback_before_after_clean(
-      ce, NR_PSTR("invokeallwith"), nr_drupal94_invoke_all_with,
-      nr_drupal94_invoke_all_with_after, nr_drupal94_invoke_all_with_clean);
+    nr_drupal8_add_method_callback_before_after_clean(
+        ce, NR_PSTR("invokeallwith"), nr_drupal94_invoke_all_with,
+        nr_drupal94_invoke_all_with_after, nr_drupal94_invoke_all_with_clean);
 #else
-  nr_drupal8_add_method_callback(ce, NR_PSTR("invokeallwith"),
-                                 nr_drupal94_invoke_all_with TSRMLS_CC);
+    nr_drupal8_add_method_callback(ce, NR_PSTR("invokeallwith"),
+                                   nr_drupal94_invoke_all_with TSRMLS_CC);
 #endif
+  }
```

Comment:
It would be easier on reviewers for the code to look like this:
```
  if (nr_drupal_hook_attribute_instrument(*retval_ptr)) {
    return;
  }
```
and leave lines 635-648 unchanged. Maybe the name `nr_drupal_hook_attribute_instrument` could be improved in this context.

Reply:
Since it's a wrapper, it'd need to be:
```
  if (nr_drupal_hook_attribute_instrument(*retval_ptr)) {
    NR_PHP_WRAPPER_LEAVE;
  }
```

Reply:
I always forget that this code is in the context of `NR_PHP_WRAPPER`. Thanks for pointing it out!

Reply:
a68bd59f4757076ea81e722c16b5cb020eb8fd4c